lib/screens/category_main_list/category_main_list_page.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import 'package:boilerplate/enums/cocktail_menu_type.dart';
+
+import 'package:boilerplate/screens/category_main_list/category_main_list_provider.dart';
+import 'package:boilerplate/screens/home/home_provider.dart';
+
+import 'package:boilerplate/widgets/bottom_navigation.dart';
+import 'package:boilerplate/widgets/cocktail_item.dart';
+import 'package:boilerplate/widgets/cocktail_list.dart';
+import 'package:boilerplate/widgets/custom_app_bar.dart';
+import 'package:boilerplate/widgets/loading_indicator.dart';
+
+class CategoryMainListPage extends ConsumerStatefulWidget {
+  const CategoryMainListPage({Key? key}) : super(key: key);
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() =>
+      _CategoryMainListPageState();
+}
+
+class _CategoryMainListPageState extends ConsumerState<CategoryMainListPage>
+    with TickerProviderStateMixin {
+  AnimationController? animationController;
+
+  @override
+  void initState() {
+    _setupController();
+    super.initState();
+
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      await ref
+          .read(categoryMainListViewModelProvider.notifier)
+          .getCategoryMainList(
+              ref.read(homeViewModelProvider.notifier).currentCocktailMenuType);
```

Comment:
Can we define this just before we use it, so that it is easier to read

Reply:
Correction is applied

![image](https://user-images.githubusercontent.com/17646275/176974991-3cbcea38-e098-425f-b3ae-7ade18b33ed5.png)